src/Autofac.Extras.MvvmCross/AutofacMvxIocProvider.cs

```diff
@@ -343,7 +378,17 @@ public void RegisterSingleton(Type tInterface, Func<object> theConstructor)
                 throw new ArgumentNullException("theConstructor");
 
             var cb = new ContainerBuilder();
-            cb.Register(cc => theConstructor()).As(tInterface).AsSelf().SingleInstance();
+
+            var type = theConstructor.GetMethodInfo().ReturnType;
+            var x = cb.RegisterType(type).As(tInterface).AsSelf().SingleInstance();
```

Comment:
There reason is, that I want to be able to call
Mvx.RegisterSingleton<IService>(Mvx.IoCConstruct<ServiceImplementation>);
By default this would not be possible, as ServiceImplementation has not been registered.
So I get the type using "theCostructor.GetMethodInfo().ReturnType" which is typeof(ServiceImplementation) and register it as itself implicitly.
That way this form of lazy initialization is also supported.